components/automate-deployment/pkg/server/server.go

```diff
@@ -1761,6 +1761,45 @@ func (s *server) ManifestVersion(ctx context.Context, d *api.ManifestVersionRequ
 	}, nil
 }
 
+// SetManifest sets the current package manifest
+func (s *server) SetManifest(ctx context.Context, req *api.SetManifestRequest) (*api.SetManifestResponse, error) {
+	if !s.HasConfiguredDeployment() {
+		return nil, ErrorNotConfigured
+	}
+
+	var (
+		err             error
+		manifestClient  manifest.ReleaseManifestProvider
+		desiredManifest *manifest.A2
+		jsonManifest    []byte
+	)
+
+	jsonManifest = req.GetManifest().GetJson()
+	if jsonManifest == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "failed to parse the manifest json")
+	}
+	manifestClient = client.NewInMemoryClient(jsonManifest)
+	desiredManifest, err = manifestClient.GetCurrentManifest(ctx, "")
```

Comment:
Yeah, that would certainly work but I sorta feel like using the clients instead of relying on the parser directly might be better. I'd rather tie us to a client interface than what the underlying parser is doing, especially as the parser and manifest evolve after time.

Reply:
I'm happy to go with y'all on this. To me, the format of the manifest is a first class citizen and we don't really gain much by intermediating our parsing via the client.